crypto/rsa/rsa_pss.c

```diff
@@ -64,9 +71,9 @@ int RSA_verify_PKCS1_PSS_mgf1(RSA *rsa, const unsigned char *mHash,
      *      -4      salt length is autorecovered from signature
      *      -N      reserved
      */
-    if (sLen == RSA_PSS_SALTLEN_DIGEST) {
-        sLen = hLen;
-    } else if (sLen < RSA_PSS_SALTLEN_AUTO_DIGEST_MAX) {
+    if (*sLen == RSA_PSS_SALTLEN_DIGEST) {
```

Comment:
I would of probably left most of this code the same and done
int sLen = *sLenOut;

*sLenOut = sLen;
err:

Reply:
Same comment applies to the add.

Reply:
Both updated.